_src/automation/generate_config_schema_export_pages.py

```diff
@@ -1,329 +1,115 @@
 
 def read_json_entries():
-	""" Load the generated JSON file and pass the entries to be read by the get_json_entries function. """
+	""" Load the generated JSON file, split into logical page chunks and generate pages. """
 
-	reference_dir = repo_root + "/reference/"
-	json_file = open(reference_dir + json_export, "r")
-	viash_json = json.load(json_file)
-	json_file.close()
+	with open(json_file, 'r') as infile:
+		viash_json = json.load(infile)
 
 	for topic in viash_json:
 		if isinstance(viash_json[topic], dict):
 			for subtopic in viash_json[topic]:
-				get_json_entries(page_title = subtopic, topic = topic, json_entry = viash_json[topic][subtopic])
+				generate_page(topic, subtopic, viash_json[topic][subtopic])
 		else:
-			get_json_entries(page_title= topic, topic = topic, json_entry = viash_json[topic])		
+			generate_page(".", topic, viash_json[topic])
 
+def generate_page(topic: str, subtopic: str, json_data: dict):
+	""" Receives JSON data, does some minor data manipulation and writes to yaml & qmd. """
 
-def get_json_entries(page_title, json_entry, topic):
-	""" Reads the generated JSON file and extract all information into instances of the JsonEntryData class. """
-
-	if topic == TOPIC_ARGUMENTS: # Keep title of argument pages as-is
-		title = page_title
+	if topic == 'arguments': # Keep title of argument pages as-is
+		title = subtopic
 	else:
-		title = clean_title(page_title)
-
-	# Sort json entries alphabetically and store in a dictionary
-	sorted_dict = sorted(json_entry, key=lambda x: x["name"], reverse=False)
-
-	for i in range(len(sorted_dict)):
-		newData = JsonEntryData()
-		newData.topic = topic
-		newData.title = title
-		newData.name = sorted_dict[i]["name"] if "name" in sorted_dict[i] else None
-		newData.examples = sorted_dict[i]["example"] if "example" in sorted_dict[i] else None
-		newData.removed = sorted_dict[i]["removed"] if "removed" in sorted_dict[i] else None
-		newData.deprecated = sorted_dict[i]["deprecated"] if "deprecated" in sorted_dict[i] else None
-		newData.since = sorted_dict[i]["since"] if "since" in sorted_dict[i] else None
-		newData.type = sorted_dict[i]["type"] if "type" in sorted_dict[i] else None
-		newData.description = sorted_dict[i]["description"] if "description" in sorted_dict[i] else None
-
-		add_json_data_to_list(newData)
-		
+		title = re.sub(r"(\w)([A-Z])", r"\1 \2", subtopic).title() # split words and capitalize
 
-def add_json_data_to_list(data : JsonEntryData):
-	""" Adds the JsonEntryData to an entry list based on its topic. """
-	if data.topic == TOPIC_FUNCTIONALITY:
-		functionality_entry_list.append(data)
-	elif data.topic == TOPIC_ARGUMENTS:
-		arguments_entry_list.append(data)	
-	elif data.topic == TOPIC_PLATFORMS:
-		platforms_entry_list.append(data)	
-	elif data.topic == TOPIC_REQUIREMENTS:
-		requirements_entry_list.append(data)
+	# Sort data entries alphabetically on 'name'
+	sorted_dict = sorted(json_data, key=lambda x: x["name"], reverse=False)
```

Comment:
The sorting of dicts may or may not work depending on the python version
I would turn this into a list of tupes (key, object) if you want to make sure it is sorted.
You can iterate over that with:

```python3

sorted_json_entries = sorted([(item_key, json_obj) for item_key, json_ob in json_data.items()], key=lambda x: x["name"])

for (key, value) in sorted_json_entries:
    ...
```

Reply:
Downside of changing inherited code; I just realized the 'sorted_dict' was a misnomer at best. It's actually a list!